binr/radare2/radare2.c

```diff
@@ -1046,6 +1046,9 @@ int main(int argc, char **argv, char **envp) {
 				while (optind < argc) {
 					pfile = argv[optind++];
 					fh = r_core_file_open (&r, pfile, perms, mapaddr);
+					if (fh && (perms & R_IO_WRITE)) {
```

Comment:
why this change? Can't I write to a file with `va` enabled?

Reply:
However, #10947 says that `io.va=false` should be done only when the file does not exist. `if (fh && (perms & R_IO_WRITE))` seems to be executed even when the file does exist, so I'm not sure about this.

Reply:
Probably just missing a bang (`!`).

Reply:
Nope, I checked and even non-existing files have a `fh` object.

Reply:
r2 helpfully creates a new file if it doesn't exist so I guess that's where a `new_file` flag should be set or something similar.

Reply:
... and it creates it before the prompt appears rather than on exit